### DISCUSSION TITLE
Evaluating code during debugging

### DISCUSSION THREAD

**Discussion by shoplab7** [wojciech-kulik/xcodebuild.nvim#266] [Q&A]
When I stop on breakpoint, how do I evaluate expressions(similar to p/po/etc in xcode)?
Also, how do I fix this issue?
> Exception: Could not find type system for language swift: TypeSystem for language swift doesn't exist

**Comment by wojciech-kulik** [comment]
In dap-ui go to repl buffer and hit `a` :). I’m not sure what’s the second issue. Maybe try a clean build from Xcode

**Reply by shoplab7** [reply]
Thanks 😅 
Regarding second issue: it seems like some issue with codelldb and it isn't going away after clean build. 
Whenever I try to evaluate any expression(even `p 1`), I'm hit with
```
[debug-adapter stderr] [ERROR codelldb::debug_session] error: Could not find type system for language swift: TypeSystem for language swift doesn't exist
error: Could not find type system for language swift: TypeSystem for language swift doesn't exist
```
Was I supposed to do anything besides downloading `.vsix` for my os(MacOS with M1 -> codelldb-darwin-arm64), unpacking it and updating codelldbPath in config?(Also, I tried printing the resulting path and it is correct)

**Reply by wojciech-kulik** [reply]
Hmm, it should be enough, you can check also:
https://github.com/wojciech-kulik/xcodebuild.nvim/wiki/Integrations#-debugger

Alternatively, try switching to the sample config and see if the issue still occurs: https://github.com/wojciech-kulik/xcodebuild.nvim/wiki/Neovim-Configuration#%EF%B8%8Fsample-config-for-ios

or maybe something is wrong with the latest codelldb release?

**Reply by shoplab7** [reply]
I had mason and it seems like it was overriding whatever xcodebuild did to dap.
I removed mason, now I'm getting
```
HOME/tools/codelldb-darwin-arm64/extension/adapter/codelldb exited with code 101
Couldn't connect to 127.0.0.1:13000: ECONNREFUSED
```
Also, when I accidentally opened `HOME/tools/codelldb-darwin-arm64/extension/adapter/codelldb` it got blocked by macos security, so I allowed it and some subsequent .py script

**Reply by shoplab7** [reply]
Tried dap/dapui configs from starter kit and codelldb 1.10.0, still getting 
```
HOME/tools/codelldb-aarch64-darwin/extension/adapter/codelldb exited with code 101
Couldn't connect to 127.0.0.1:13000: ECONNREFUSED
```
I tried starting it in terminal myself with
```
Path/codelldb --port 13000
```
nvim was able to connect to it, but when I tried to evaluate an expression, I got
```
[ERROR codelldb::debug_session] error: Could not find type system for language swift: TypeSystem for language swift doesn't exist
```

**Reply by shoplab7** [reply]
Found the issue - I'm using xcodes, so my path to xcode is `/Application/Xcode-16.1.0.app/` instead of `/Applications/Xcode.app`
After I tried changing it in https://github.com/wojciech-kulik/xcodebuild.nvim/blob/157197db1c701b6b3fc340f6d8213aa4a1fa07ac/lua/xcodebuild/integrations/dap.lua#L416, everything started working fine. So, I reverted it and created symlink for xcode.

---

You might also use `xcode-select --print-path` to determine the path to Xcode before using `/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Versions/A/LLDB` as the path to LLDB.
